### DISCUSSION TITLE
Celery configuration, DAG distribution across machines

### DISCUSSION THREAD

**Discussion by MattiaGallegati** [apache/airflow#23500] [Q&A]
Hello there,
I'm relatively new to Airflow.
I'm planning to adopt it in a production environment using CeleryExecutor.
As I could understand you need to distribute DAG across the machines that will use them. In order to do that we are supposed to manually copy dags or sync them via Git or other methods.
Is this actually necessary?
Won't Airflow provide a convenient way to automatically distribute DAG from the scheduler machine to worker nodes (using some kind of serialization)?
Should I open a feature request for that or am I missing something?

**Comment by potiuk** [comment]
Yes it is necessary. Distributing DAGs is a deployment job not Airflow. 

There were many discussions in the past on different way of distributing them and so far they resulted in those two AIPS:

* https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-5+Remote+DAG+Fetcher
* https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-20+DAG+manifest

But they are quite old, not really discussed for quite some time, and general conclusion is that implementing somethign that will be generic and will work in all ways of deploying airflow, with multiple DAGs and dynamic nature of Python, this might be either extremely difficult to do better than some way of just distributing the files via available deployment options. 

If you wish, you are free (like anyone else) to revive the discussion and raise it at the devlist. It's far too huge of a change to be discussed in "feature request" as it fundamentally changes Airflow's architecture.

But - as everything in open-source - it will need someone to lead the effort, discussion, and eventually implementation and testing of such a feature (and I can tell you it's huge). If you wish to take on such task -feel free - you will find some past discussions at the links there.

**Reply by MattiaGallegati** [reply]
Thank you for your kind response and for your work on the platform,

It was just a question from a noob who is trying to figure out some aspects of this complex ecosystem.
Thank you,
